extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -220,6 +284,71 @@ export class AmpSelector extends AMP.BaseElement {
   }
 
   /**
+   * Handles click events for the selectables.
+   * @param {!Event} event
+   */
+  clickHandler_(event) {
+    let el = dev().assertElement(event.target);
+    if (!el) {
+      return;
+    }
+    if (!el.hasAttribute('option')) {
+      el = closest(el, e => e.hasAttribute('option'), this.element);
+    }
+    if (el) {
+      this.onOptionSelected_(el);
+    }
+  }
+
+  /**
+   * Handles keyboard events for the selectables.
+   * @param {!Event} event
+   */
+  keyDownHandler_(event) {
+    // Make currently selected option unfocusable
+    this.options_[this.focusedIndex_].tabIndex = -1;
+
+    const dir = this.win.document.body.getAttribute('dir') || 'ltr';
+    const ltr = dir != 'rtl';
+    let delta = 0;
+
+    switch (event.keyCode) {
+      case 37: // Left
+        delta = ltr ? -1 : 1;
+        break;
+      case 38: // Up
+        delta = -1;
+        break;
+      case 39: // Right
+        delta = ltr ? 1 : -1;
+        break;
+      case 40: // Down
+        delta = 1;
+        break;
+    }
+
+    if (delta == 0) {
+      return;
+    }
+    event.preventDefault();
+
+    this.focusedIndex_ = (this.focusedIndex_ + delta) % this.options_.length;
```

Comment:
add some comments describing what is going on here. 

Also could you explain what the focusedIndex_ is used for? 

It would be great if we can have a single method modifying tabIndex/focus on all elements based on any scenario. Do you think that is possible ?